CppSamples/ChilliConnectCppDemo/ChilliConnect.cpp

```diff
@@ -0,0 +1,122 @@
+#include "ChilliConnect.h"
+
+ChilliConnect::ChilliConnect(const string gameToken)
+{
+	this->gameToken = gameToken;
+	this->httpSystem = unique_ptr<HttpSystem>(new HttpSystem());
+}
+
+ChilliConnect::~ChilliConnect()
+{
+}
+
+bool
+ChilliConnect::parse(const HttpResult &httpResult, ChilliConnectResponse & response, Json::Value & json)
+{
+	response.wasOk = (httpResult.code == 200);
+	response.rawBody = httpResult.body;
+
+	bool parsingSuccessful = false;
+	if (!httpResult.body.empty()) {
+		Json::Reader reader;
+		parsingSuccessful = reader.parse(httpResult.body, json);
+		if (!response.wasOk) {
+			response.errorCode = json["Code"].asInt();
+			response.errorMessage = json["Error"].asString();
+		}
+	}
+
+	return parsingSuccessful;
+}
+
+CreatePlayerResponse
+ChilliConnect::CreatePlayer() 
+{
+	//https://docs.chilliconnect.com/api/?system=http#api-PlayerAccounts-CreatePlayer
+
+	map<string, string> headers;
+	headers["Game-Token"] = "Vv7VANzImRtEUeiYaoz4lWKqB6t349iy";
+
+	HttpResult httpResult = this->httpSystem->MakePostRequest("connect.chilliconnect.com", "1.0/player/create", headers, "");
+
+	CreatePlayerResponse response;
+	
+	Json::Value json;
+	if (parse(httpResult, response, json) && response.wasOk) {
+		response.chilliConnectId = json["ChilliConnectID"].asString();
+		response.chilliConnectSecret = json["ChilliConnectSecret"].asString();
+	}
+
+	return response;	
+}
+
+LoginResponse
+ChilliConnect::Login(const string chilliConnectId, const string chilliConnectSecret)
```

Comment:
Most of the string arguments in this class are passed by value and not reference (should prob change to const string& chilliConnectId